sql/modules/Voucher.sql

```diff
@@ -365,6 +365,13 @@ $$
         WHERE id IN (select trans_id FROM voucher
                 WHERE batch_id = in_batch_id);
 
+        -- When approving the AR/AP batch import,
+        -- we need to approve the acc_trans line also.
+        UPDATE acc_trans SET approved = true
+        WHERE trans_id IN (select trans_id FROM voucher
+                WHERE batch_id = in_batch_id
+                AND batch_class IN (1, 2));
```

Comment:
Can't the query below be expanded by batch classes 1 and 2? (so we have shorter, more readable, code?)

Reply:
(At the same time, I'm wondering why the original author would not have included batch types 1 and 2 from the start...)

Reply:
> Can't the query below be expanded by batch classes 1 and 2? (so we have shorter, more readable, code?)

The original author write the below query is for payment/receipt batch. which join acc_trans and voucher on voucher_id.
However, the AR/AP Batch import doesn't set the voucher_id, it join voucher with trans_id. That's why I add separate query for that.

Reply:
> (At the same time, I'm wondering why the original author would not have included batch types 1 and 2 from the start...)

If I remember correctly, the lines in acc_trans are inserted as approved when save draft. But in 1.8, lines are inserted as unapproved when save draft and then update approved when Post or approve  from Transaction Approval.

Reply:
> The original author write the below query is for payment/receipt batch. which join acc_trans and voucher on voucher_id.
> However, the AR/AP Batch import doesn't set the voucher_id, it join voucher with trans_id. That's why I add separate query for that.

I found a comment about this here.
https://github.com/ledgersmb/LedgerSMB/blob/009cdf6c6ec2b7ac0ebc3123047d67207cf2e6fc/sql/modules/Voucher.sql#L439-L440

Reply:
> If I remember correctly, before 1.8, the lines in acc_trans are inserted as approved when save draft. But in 1.8, lines are inserted as unapproved when save draft and then update approved when Post or approve from Transaction Approval

Correct. I think in 1.9 that is one of the things we changed.

Reply:
Do you need more clarification? Or Shall we mark as resolve?

Reply:
Marking resolved